src/shared/redux/modules/member.js

```diff
@@ -37,18 +40,30 @@ const reducer: Reducer<State, Action> =
       case DEACTIVATED_MEMBER:
         return (
           { ...member
-          , activation_status: { value: 'deactivated', initial_value: member.activation_status.initial_value }
+          , activation_status: { value: 'deactivated', initialValue: member.activation_status.initialValue }
           , deletion_date: { value: new Date().toISOString() }
           }
         )
       case REACTIVATED_MEMBER:
         return (
           { ...member
-          , activation_status: { value: 'activated', initial_value: member.activation_status.initial_value }
+          , activation_status: { value: 'activated', initialValue: member.activation_status.initialValue }
           , deletion_date: {}
           , deletion_reason: { value: null }
           }
         )
+      case UPDATED_MEMBERSHIP_TYPE:
+        const newState =
+          { ...member
+          , membership_type: { ...member.membership_type, value: payload }
+          , date_membership_type_changed: payload !== member.membership_type.initialValue
+            ? { value: format_date(new Date()), initialValue: (member.date_membership_type_changed && member.date_membership_type_changed.initialValue) || null }
```

Comment:
No, I mean how you don't spread `member.date_membership_type_changed` like you do if the conditional is false